src/workers/userStreakUpdatedCio.ts

```diff
@@ -21,20 +21,49 @@ const worker: TypedWorker<'api.v1.user-streak-updated'> = {
       return;
     }
 
+    const user: Pick<User, 'timezone'> | null = await con
+      .getRepository(User)
+      .findOne({
+        select: ['timezone'],
+        where: {
+          id: streak.userId,
+        },
+      });
+
+    if (!user) {
+      return;
+    }
+
     const readHistory = await getUserReadHistory({
       con,
       userId: streak.userId,
       before: new Date(),
       after: subDays(new Date(), 7),
     });
-    const readHistoryDates = readHistory.flatMap((item) => item.date);
+    const readHistoryDates = readHistory.flatMap((item) =>
+      formatInTimeZone(
+        item.date,
+        user.timezone || DEFAULT_TIMEZONE,
+        'yyyy-MM-dd',
+      ),
+    );
 
     const lastSevenDays = [...Array(7)].reduce((acc, _, i) => {
-      const d = new Date();
-      d.setDate(d.getDate() - (6 - i));
+      const dateOfWeek = subDays(new Date(), 6 - i);
+
+      const dStamp = formatInTimeZone(
+        dateOfWeek,
+        user.timezone || DEFAULT_TIMEZONE,
+        'yyyy-MM-dd',
+      );
+      const day = formatInTimeZone(
+        dateOfWeek,
+        user.timezone || DEFAULT_TIMEZONE,
+        'iiiiii',
+      );
       acc.push({
-        day: days[d.getDay()],
-        read: readHistoryDates.includes(d.toISOString().split('T')[0]),
+        day,
+        read: readHistoryDates.includes(dStamp),
       });
```

Comment:
From what I see in CIO and digest emails when we show reading streak we are expecting local days per timezone, but currently for timezones not close to UTC the day would basically be wrong as it was on frontend. 

Since I am not sure CIO snippets have the ability to manipulate timezones we continue to do that only know it should be correct for all timezones.

Reply:
Top of head the CIO snippet actually takes user timezone and offsets to UTC should be ok there.
But this is top of head.

Reply:
Yeah it does not, but new logic should be alright. I'll test by moving myself to another TZ.